src/solidity/stateDecoder.js

```diff
@@ -0,0 +1,79 @@
+var decoder = require('./stateDecoder')
+var astHelper = require('./astHelper')
+var typeDecoder = require('./typeDecoder')
+
+/**
+  * decode the contract state storage
+  *
+  * @param {Array} storage location  - location of all state variables
+  * @param {Map} storageContent  - storage
+  * @return {Map} - decoded state variable
+  */
+function decodeState (storageLocations, storageContent) {
+  if (storageContent['0x']) {
+    storageContent['0x00'] = storageContent['0x']
+    storageContent['0x'] = undefined
+  }
+  var ret = {}
+  for (var k in storageLocations) {
+    var stateVar = storageLocations[k]
+    var location = {
+      slot: stateVar.location.slot,
+      offset: stateVar.location.offset
+    }
+    ret[stateVar.name] = decoder.decode(stateVar.type, storageContent, location)
+  }
+  return ret
+}
+
+/**
+  * return all storage location variables of the given @arg contractName
+  *
+  * @param {String} contractName  - name of the contract
+  * @param {Object} astList  - AST nodes of all the sources
+  * @return {Object} - return the location of all contract variables in the storage
+  */
+function getStateVariableLocations (contractName, astList) {
+  var stateDefinitions = astHelper.getStateDefinition(astList, contractName)
+  var ret = []
+  if (!stateDefinitions) {
```

Comment:
if ´getStateDefinition´ returns null, it means that something went wrong (basically it did not find the contract in the astlist).
it's more how can we handle a failure of ´getStateDefinition´ here. 
I would say that we could do that in another PR (having a module that can handle error and gives feedback to the user)